src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1183,6 +1269,35 @@ private void CreateNextSegment(string text)
             this.CreateDynamicPathSegment(previous, identifier, parenthesisExpression);
         }
 
+        private bool TryCreateEscapeFunctionSegment(string segmentText)
+        {
+            int numberOfSegmentsParsed = this.parsedSegments.Count;
+            string newSegmentText = segmentText.Substring(0, segmentText.Length - 1);

```

Comment:
>string newSegmentText = segmentText.Substring(0, segmentText.Length - 1); [](start = 12, length = 73)

this line is abrupt. Maybe add a comment or a Debug.Assert(...) to make sure the input segment text has a ":" at the end?